### PR TITLE
Display Imminent instead of zero or negative days for Cluster Deletion in UI

### PR DESCRIPTION
**Pull request by pedrottimark**
## Description

Here is a screen grab with temporary code change to replace `null` with `{ daysUntilDeletion: 0 }` for `clusterRetentionInfo` prop of `ClusterDeletion` element:
Imminent

Also I corrected my misunderstanding about `isExcluded` property.

### Changed files

1. Edit cypress/integration/clusters/clusterDeletion.test.js
    * Add 2 tests for zero and negative days
    * Delete test for `isExcluded: false` which is not a valid case

2. Edit src/Containers/Clusters/Components/ClusterDeletion.tsx
    * Correct the comments
    * Add **Imminent** for `daysUntilDeletion < 1`
    * Delete **Deletion is turned off** for `isExcluded: false` which is not a valid case

3. Edit src/types/clusterService.proto.ts
    * Correct the comments

### Residue

1. Improve integration tests to mock response in clusters table instead of cluster panel, because **Cluster Deletion** is below the fold on the panel. If the test fails, the snapshot displays the incorrectly rendered result. Also possible to use test to take pictures of UI for different scenarios.

## Checklist
- [x] Investigated and inspected CI test results
- [x] Edited integration tests

## Testing Performed

1. `export ROX_DECOMMISSIONED_CLUSTER_RETENTION=true` in ui
2. `yarn deploy-local` in ui
3. `yarn start` in ui
4. `export CYPRESS_ROX_DECOMMISSIONED_CLUSTER_RETENTION=true` in ui/apps/platform
5. `yarn cypress-open` in ui/apps/platform
    * clusters/clusterDeletion.test.js see that existing and new tests pass

### CODE REVIEW

File: central/cluster/service/service_impl.go [hunk 1]
```diff
@@ -141,6 +141,11 @@ func (s *serviceImpl) getClusterRetentionInfo(ctx context.Context, cluster *stor
 		return nil, nil
 	}
 
+	// If retention is disabled, there is no "days remaining" calculation to be done.
```
Comment (charmik-redhat) [inline central/cluster/service/service_impl.go:144]: Can we also short circuit the `getClusterIDToRetentionInfoMap` function when retentionDays == 0 ?

Comment (theencee) [reply central/cluster/service/service_impl.go:144]: I don't think it makes a massive difference since the beginning of this method doesn't really do much. And it's cleaner than leaking the config outside of this method. Thoughts?

**Review by charmik-redhat — APPROVED** [review]

File: ui/apps/platform/cypress/integration/clusters/clusterDeletion.test.js [hunk 2]
```diff
@@ -92,17 +92,24 @@ describe('Cluster page clusterRetentionInfo', () => {
...
+    it('should display Imminent for -1 days in Cluster Deletion widget for mock request', () => {
+        const clusterRetentionInfo = { daysUntilDeletion: -1 };
+        visitClusterWithRetentionInfo(clusterRetentionInfo);
+
+        cy.get('div:contains("Cluster Deletion"):contains("Imminent")'); // FYI red color
     });
 
-    it('should display Deletion is turned off in Cluster Deletion widget for mock request', () => {
-        const clusterRetentionInfo = { isExcluded: false };
+    it('should display Excuded from deletion in Cluster Deletion widget for mock request', () => {
```
Comment (vjwilson) [inline ui/apps/platform/cypress/integration/clusters/clusterDeletion.test.js:109]: nit: typo
```suggestion
    it('should display Excluded from deletion in Cluster Deletion widget for mock request', () => {
```

File: ui/apps/platform/src/Containers/Clusters/Components/ClusterDeletion.tsx [hunk 3]
```diff
@@ -24,19 +24,23 @@ function ClusterDeletion({ clusterRetentionInfo }: ClusterDeletionProps): ReactE
...
         const healthStatus = getClusterDeletionStatus(daysUntilDeletion);
         const { bgColor, fgColor } = healthStatusStyles[healthStatus];
-        const text = daysUntilDeletion === 1 ? 'in 1 day' : `in ${daysUntilDeletion} days`;
+        /* eslint-disable no-nested-ternary */
+        const text =
+            daysUntilDeletion < 1
+                ? 'Imminent'
+                : daysUntilDeletion === 1
+                ? 'in 1 day'
+                : `in ${daysUntilDeletion} days`;
```
Comment (vjwilson) [inline ui/apps/platform/src/Containers/Clusters/Components/ClusterDeletion.tsx:33]: I personally think a `switch` is easier to read than nested ternary, but 1 extra level, as it is here, it's ok.

**Review by vjwilson — APPROVED** [review]
